### DISCUSSION TITLE
How to Change Password of User?

### DISCUSSION THREAD

**Discussion by goldcoders** [supabase/supabase#697] [Questions]
GoTrue password is hash during creation, but how do u update the password
assuming the user change the password, NOT RESET , since the reset  and email confirmation is unworkable atm.

Is there a built un hash function in postgresql?

the auth is a function on the postgresql youve added.

Im wondering if I can put a watcher on auth.user tables 
if the password changes it will Re-Has the New password...

**Comment by kiwicopple** [comment]
Hi @goldcoders, the password change should *always* be done through a reset email. It should be handled by the user themselves, for security reasons.

> NOT RESET , since the reset and email confirmation is unworkable atm.

It seems like this is the main issue - can you clarify what isn't working? Are the emails going into spam? If they are, could you try to use a custom SMTP server?

FWIW, the password hashing is managed by the auth server, GoTrue: https://github.com/supabase/gotrue/blob/master/api/recover.go

**Reply by AIQMEDIA** [reply]
In the password recovery for flutter, the email doesn't redirect back to the app and shows error 404. Cn you please tell me how to resolve this ?

**Reply by tawjaw** [reply]
@kiwicopple changing password must be done with reset email but a signed in user can change password without a reset email right? 

Does the reset email give any permissions for updating a password?
